packages/now-cli/src/util/dev/server.ts

```diff
@@ -762,10 +769,28 @@ export default class DevServer {
     // Retrieve the path of the native module
     const nowConfig = await this.getNowConfig(false);
     const nowConfigBuild = nowConfig.build || {};
-    const [runEnv, buildEnv] = await Promise.all([
+
+    const cloudEnvVariables = await getDecryptedEnvRecords(
+      this.output,
+      client,
+      project,
+      4,
+      ProjectEnvTarget.Development
+    );
+
+    const [localEnvVars, localBuildEnvVars] = await Promise.all([
       this.getLocalEnv('.env', nowConfig.env),
       this.getLocalEnv('.env.build', nowConfigBuild.env),
     ]);
+
+    const runEnv = Object.keys(localEnvVars).length
+      ? localEnvVars
+      : cloudEnvVariables;
+    const buildEnv = Object.keys(localBuildEnvVars).length
+      ? localBuildEnvVars
+      : cloudEnvVariables;
+
+    // if local .env/.env.build file exists, don't use cloud variables
     const allEnv = { ...buildEnv, ...runEnv };
```

Comment:
I think you need to change this logic because the build/run env vars only makes sense if there are files. If you are using the cloud, then it should be all or nothing.

Try something like this instead:

```js
let allEnv = { ...buildEnv, ...runEnv };
if (Object.keys(allEnv).length === 0) {
  allEnv = await getDecryptedEnvRecords(/*...*/);
}
```